### ISSUE TITLE
Analog nudging still broken, from mid 1.7.0 beta to present

### ISSUE THREAD

**Issue by wiesshund** [vpinball/vpinball#106]
All roads lead to UP

tested with Xinput device (dof nudge preset set to xinput API)
https://i.imgur.com/68DsKL5.png

Apron indicators as well as numeric display show correct axis inputs
but balls only react by going UP
and it does not matter how strong the nudge is either, the balls go up at the same force regardless
in 10.6.X if you set the nudge hard enough you can nearly drive the balls around the table
in 10.7.X all you can do by turning nudge up is tilt yourself faster, the balls still just blip UP a little

Video below will explain the issue fully

www.youtube.com/watch?v=8aS8zWe6jM0

if switched to dinput and switched to a dinput device, the nudging will work, but has to be totally recalibrated
but the feedback breaks
giving vpx a set of dinput dlls seems to do something to feedback (stick vibrates when table loads)
but then the nudging ceases working right

this was reported during 10.7 beta but at the time we didnt have github to log bugs

**Comment by Chickenzilla** [comment]
Yup, I experienced this too.  #77 also seems to be a duplicate issue.  I'm on linux so building/validating is not a simple thing for me to do (the editor barely works in wine/proton), but I'm pretty sure I found the bug for any other devs who want to try to tackle this.

In Player::NudgeUpdate(), line https://github.com/vpinball/vpinball/blob/master/pin/player.cpp#L2411 iterates over the number of joysticks detected by PinInput.  In PinInput, `e_JoyCnt` is only updated on one line, inside DIEnumJoystickCallback on line https://github.com/vpinball/vpinball/blob/master/pininput.cpp#L322.  This is our problem, because DIEnumJoystickCallback is only invoked via DInput::EnumDevices, which is only kicked off if m_inputApi == 0 on line https://github.com/vpinball/vpinball/blob/master/pininput.cpp#L883.  Note that m_inputApi is 1 when XInput is enabled (and support is compiled in).

I'm not familiar enough with this codebase to suggest the most solid fix.  My naive guess would be to set e_JoyCnt to 1 in PinInput::handleInputXI on/around line https://github.com/vpinball/vpinball/blob/master/pininput.cpp#L586, where you know you have a valid XInput device.  This should work because this code only ever seems to register the first XInput device as `APP_JOYSTICK(0)`, so everything should expect data in the output from that joystick to exist.

Curiously, the reason the plumb "works" on the testing table is because it doesn't read from the updated nudge variables, it just directly reads the accel values from the first joystick provided (the one XInput already exclusively updates).  Not sure why the hacks around only supporting one, but I'm not here to complain about this cool piece of software.  Hope this helps!
